lib/bloc/pos_catalog/bloc.dart

```diff
@@ -32,6 +34,31 @@ class PosCatalogBloc with AsyncActionsHandler {
     });
     listenActions();
     _currentSaleController.add(Sale(saleLines: List()));
+    _trackCurrentSaleRates(accountStream);
+  }
+
+  void _trackCurrentSaleRates(Stream<AccountModel> accountStream){
+    accountStream.listen((acc) {
+      var currentSale = _currentSaleController.value;
+
+      // In case the price is locked we don't calculate the charge.
+      if (currentSale.priceLocked) {
+        return;
+      }
+      _currentSaleController.add(currentSale.copyWith(
+        saleLines: currentSale.saleLines.map(
+          (sl){
+            double rate = sl.satConversionRate;
+            Currency curr = Currency.fromTickerSymbol(sl.currency);
```

Comment:
CurrencyWrapper.fromShortName(sl.currency, acc) can be used here.

Reply:
Right now we can't since currently the CurrencyWrapper is only at the UI level. Once we move it into the bloc we can use it here.